packages/test/test-end-to-end-tests/src/test/gc/gcLoadingFromOlderSummaries.spec.ts

```diff
@@ -203,6 +225,80 @@ describeNoCompat("GC loading from older summaries", (getTestObjectProvider) => {
             dsBReferenceState3 === false, `dataStoreB should still be unreferenced on loading from old summary`);
     });
 
+    /**
+     * In this test, the data store containing references to another data store is changed after loading from an older
+     * summary. But the DDS in the data store containing the references is not changed. This validates that the GC data
+     * in older summary is correctly propagated to DDS as well.
+     */
+    it("updates unreferenced nodes correctly when DDS is unchanged after loading from older summary", async () => {
+        const summarizer1 = await createSummarizer(provider, mainContainer);
+
+        // Create a second DDS in dataStoreA. This will be changed after loading from old summary so that the data store
+        // changes but the root DDS containing references is unchanged.
+        const dataStoreAdds2 = SharedMap.create(dataStoreA._runtime);
+        dataStoreA._root.set("dds2", dataStoreAdds2.handle);
+
+        const dataStoreBHandle =
+            (await containerRuntime.createDataStore(TestDataObjectType)).entryPoint as IFluidHandle<ITestDataObject>;
+        assert(dataStoreBHandle !== undefined, "New data store does not have a handle");
+        const dataStoreB = await dataStoreBHandle.get();
+        dataStoreA._root.set("dataStoreB", dataStoreBHandle);
+        dataStoreA._root.delete("dataStoreB");
```

Comment:
What effect does setting and deleting have?

Reply:
It makes the new data store unreferenced in the summarizer client. We need to add the handle first so that the data store becomes live and the summarizer client knows about its existence.